src/summary/patient-chart-summary.component.tsx

```diff
@@ -1,17 +1,49 @@
 import React from "react";
 import { RouteComponentProps } from "react-router";
 import styles from "./patient-chart-summary.css";
-import HistorySection from "./history/history-section.component";
-import DocumentationSection from "./documentation/documentation-section.component";
+import DimensionsCard from "./documentation/dimensions-card.component";
+import VitalsCard from "./documentation/vitals-card.component";
+import ConditionsCard from "./history/conditions-card.component";
+import AllergyCard from "./history/allergy-card.component";
+import NotesCard from "./history/notes-card.component";
+import ProgramsCard from "./history/programs/programs-card.component";
 
 export default function PatientChartSummary(props: PatientChartSummaryProps) {
+  const config = [
+    "conditions",
+    "programs",
+    "allergies",
+    "notes",
+    "vitals",
+    {
+      module: DimensionsCard
+    }
+  ];
+
+  const coreComponents = {
+    conditions: ConditionsCard,
+    programs: ProgramsCard,
+    allergies: AllergyCard,
+    notes: NotesCard,
+    vitals: VitalsCard,
+    heightAndWeight: DimensionsCard
+  };
+
   return (
     <main className="omrs-main-content">
-      <div className={styles.patientSummary}>
-        <HistorySection match={props.match} />
-      </div>
-      <div className={styles.patientSummary}>
-        <DocumentationSection match={props.match} />
+      <div className={styles.h}>
+        <div className={styles.patientChartCards}>
+          {config.map((widget, index) => {
+            let Component;
+            if (typeof widget === "string") {
+              Component = coreComponents[widget];
+            } else {
+              Component = widget["module"];
+            }
```

Comment:
I suggest something like

```
const coreComponents = {
  "notes": NotesComponent,
  ...
}
const otherComponentArray = config.widgetDefinitions.map(dfn => {
  const module = SystemJS.import(dfn.esModule)
  const component = module[dfn.exportName]  // idk
  return { key: dfn.name, value: component }
})
const otherComponents = // massage otherComponents from an array of objects with "key" and "value" keys into a single object with the value of "key" as the keys and the value of "value" as the values
const components = coreComponents.mergeOrWhatever(otherComponents)  // maybe it's 
assign"?
config.map((widgetName, index) => {
  const Component = components[widgetName]
  return <Component props={props.match} key={index} />;
})
```

Reply:
As described above, I'd like to keep as is for now. Once config module supports arrays (and hopefully dynamic loading of es6 modules), I will refactor code as necessary.